### ISSUE TITLE
State complication shows Icon instead

### ISSUE THREAD

**Issue by gaardiolor** [Kemmey/Kemmey-TeslaWatch-Public#226]
Hello,

I've updated the watch app to v1.1.0 . Since that update, after some time, the 'State' complication switches to the 'Icon'. Or well, it's still the 'State' complication, but it's displaying the icon. 

This happens:
- After some time of configuring the 'State' complication to be displayed on the Watch face
- Each time I'm raising my wrist. When I'm not raising my wrist, it's still displaying the State information
- If I switch the complication to 'Icon' and back to 'State', it's fixed for a few hours / days, then it breaks again. 

![image](https://user-images.githubusercontent.com/22920143/100473221-25347580-30de-11eb-8dd0-b89183ba9d7b.png)

**Comment by Kemmey** [comment]
Hey @gaardiolor 

Yeah, same issue recorded here:

https://github.com/Kemmey/Kemmey-TeslaWatch-Public/issues/219

It's super weird! Been trying to reproduce for past ten days, no luck :-(

I really have no idea what is going on. 

Is this only for the large graphic rectangular complication?

**Comment by gaardiolor** [comment]
Ah, not sure how I missed that other issue.. I did search, but not good enough apparently. 

I’ve configured a new face with state also in the small round complication, I will let you know!

**Comment by gaardiolor** [comment]
Ok @Kemmey , it's only a problem (at least up until now) with the rectangular one. 

![image](https://user-images.githubusercontent.com/22920143/100515806-fa4e2e00-317e-11eb-9d84-5ab450e1c593.png)

**Comment by gaardiolor** [comment]
Also, if I have the rectangular 'State' on 2 faces, simply switching one of them to 'Icon' and back to 'State' doesn't fix it. 

I have to set the first watch face to 'Icon', go to the second face, switch it to 'Icon' and back to 'State', then it's fixed.

**Comment by Kemmey** [comment]
@gaardiolor Don't sweat the double post - would much rather have more posts than none!

When you do the switch to icon, how long is it fixed? Hours or days?

**Comment by gaardiolor** [comment]
I'll note timestamps and look if there's a pattern. I'd say at most a few days, but not sure. Most of the times once per day. 

Never wrote an apple app, but perhaps the complicationdata is accessed using an array that's changed.. or an object that's overwritten. In the past I've never had this problem, it must be either v1.1.0 or the companion app that changed something.

**Comment by Kemmey** [comment]
The change was introduced with 1.0.8 where complication actions were added. Before that the status widget was the only widget. But now, somehow, the OS confuses the id of the status widget with that of the icon widget. The configuration is heavily charged, even across reboots and reinstalls. As this feature is new to watchOS 7, it might possibly be an OS bug. I certainly can't reproduce it. 

Doh, replace widget with complication above :-D

**Comment by gaardiolor** [comment]
So far, the times when I first noticed the 'Icon' instead of 'State':
Fri 27 nov 18:05
Sat 28 nov 13:21
Sun 29 nov 14:57
Mon 30 nov -  No issue
Tue 1 dec 14:17

**Comment by gaardiolor** [comment]
Maybe relevant since you mentioned the caching behaviour persisting across reinstalls, and quite 'rare' so perhaps something that explains it only happening to a few people: Before the first occurance I did uninstall and reinstall the app. 

<sub>Reinstall was in the process of trying to get the iphone companion app to install (in the app store the app was showing as 'purchased' without an install option - fixed in the end by installing some random other app).</sub>

**Comment by Kemmey** [comment]
Hey - just an update - I'm really sorry that I've been unable to get any traction on this issue. I just really have zero clue as to what's going on. My best bet is still that this must be a watchOS bug - though that doesn't help you :-(

I'm still trying out things to see if I can replicate the issue, but have so far been unsuccessful.

**Comment by gaardiolor** [comment]
ok, just some last dates, just to show that sometimes it can happen multiple times per day, sometimes it can skip a day:
Wed 2 dec 00:13
Wed 2 dec 17:42
Sat 5 dec 14:58
Sun 6 dec 14:57
Sun 6 dec 22:59
Mon 7 dec 13:03
Mon 7 dec 18:19

I can't really seem to find a pattern. Anything I can do / test ?

**Comment by gaardiolor** [comment]
@Kemmey I have unpaired, erased and reinstalled my watch ( restored from backup, but apps did reinstall ), but the problem came back the next day. I did however notice the Icon as well before selecting my car in the watch app. So apparently the app has some logic to sometimes display the Icon in the Stats complication.

**Comment by gaardiolor** [comment]
Last time the issue happened was on dec 12th, one week ago. Since it started this is a new record. Maybe (the update to) WatchOS 7.2 fixed it. If it's still fixed in one week I'll close this issue.

**Comment by gaardiolor** [comment]
I jinxed it.. issue is back.

**Comment by mallo3w** [comment]
I've got the same issue. Bug occurs multiple times per day for me. Changing the complication to 'Icon' then back to 'State' again seems to fix it, until next time, that is!

**Comment by Tesla3awd2018** [comment]
Having same problem on my Apple Watch series 3, but I sometimes get a "(lightening bolt symbol) followed by 75%-sentry on" for States which is not correct and then it goes back to icon mode.

**Comment by Flybuddy** [comment]
Having same problem for several weeks, it seemed to have fixed itself to where it only did it every few days but now is back with a vengeance and consistently occurring about 3 times daily.

**Comment by Flybuddy** [comment]
Still occurring 2 or 3 times a day. Any chance of just getting rid of the icon symbol altogether in next update?

**Comment by Kemmey** [comment]
@Flybuddy this is such an annoying bug - no matter what I do, I can't reproduce it, so I can't figure out what's wrong. 

My best guess is a memory issue of some sort - or potentially a broken cache, though that should have fixed itself long ago.

**Comment by Flybuddy** [comment]
Thx for the follow up. I noticed on github that no one else seems to have complained about it for the past several weeks. Am I the only one still having the problem? If so, it might be worthwhile for me to delete and reload again. Didn’t work before but who knows. Is the Tesla symbol icon needed? Maybe if it wasn’t there it couldn’t jump over. No matter what, it’s a great app and I can live with the minor inconvenience. Take care


Sent from Yahoo Mail for iPad


On Saturday, March 6, 2021, 3:45 PM, Kim Hansen <notifications@github.com> wrote:




@Flybuddy this is such an annoying bug - no matter what I do, I can't reproduce it, so I can't figure out what's wrong.

My best guess is a memory issue of some sort - or potentially a broken cache, though that should have fixed itself long ago.

**Comment by smakson11** [comment]
I'm still having the issue.

I just bought it a few days ago, but without this working I don't want it.

**Comment by Kemmey** [comment]
Unfortunately I'm nowhere near to a solution for this - I've tried all I can think of, and am unable to reproduce. I know this is happening, but all I can do is chalk this one up as a bug in watchOS 7 that apple will have to fix.

**Comment by smakson11** [comment]
I've moved over to the modular face for now.  Keep us posted.

**Comment by SGS68** [comment]
I just bought your app today and the modular face is the only one I use on my Apple Watch 6 running the very latest iOS as of 3/29/2021.  The large square interface in the middle of my watch is what I want to use for the Watch for Tesla complication. I too have the same issue. I keep my wrist at an angle and I can see my car stats. As soon as I turn my wrist to my face (watch raise) it changes over to the big watch for Tesla red logo only.  I’m day 1 on the app and like the functionality but not this bug.  Shawn

**Comment by SGS68** [comment]
![AF28182B-2A10-497C-BA5E-335D0B781EEC](https://user-images.githubusercontent.com/81597139/112913723-4a56e600-90c8-11eb-8e37-3d92c771efd4.jpeg)
![CDB3081D-CF58-42F5-A219-514FB25EA2AE](https://user-images.githubusercontent.com/81597139/112913724-4aef7c80-90c8-11eb-93bd-81e03bbd56f8.jpeg)

**Comment by gaardiolor** [comment]
@Kemmey There must be some narrowing down we can do. For example, create an Icon2 complocation, see if it changes to Icon2, could give a hint. Or some sort of logging, maybe it defaults to Icon if there's an error? It might be an apple bug, but I can't find any other apps that have this issue. If it's an Apple bug you must be doing something pretty special, which could maybe be solved by rewriting an exotic piece of code ?

**Comment by SGS68** [comment]
Duhhh.. Okay I fixed it.  I went in and looked at the complication again
after I hit edit.  It lets me pick between either showing the icon, or the
state of the car.  I now picked "State" and it's fine.  I guess only if you
pick to just show the icon does it have the weird glitch when you are not
raised wrist and looking at an angle it shows you the state of the car
view.

**Comment by smakson11** [comment]
I am pretty sure everybody here has it on state.  It will work for awhile, and then will be bad again.

**Comment by SGS68** [comment]
Yes I spoke to soon. Later this afternoon I look down and it’s back to showing just the icon instead of the state of the car. Only thing I can think of I did was take my watch off for a while and had to enter my 4 digit number security passcode when I put it back on.  When I went to edit and fix it back it was on state and not icon so it was showing wrong.

V/r,
Shawn

Shawn Savchenko
Cell:  (919) 333.6285
***@***.***

**Comment by PhotonHebs** [comment]
Hey Kemmey, I am also having this issue.  Here is a link to a video showing the problem: https://www.dropbox.com/s/4pf0z8j0pba7a9m/Video%2007-04-2021%2C%2013%2000%2040.mov?dl=0

Watch OS 7.3.1
iOS 14.4.2

Editing the watch face fixes it for an hour or so but then it reverts back to the issue again.  I have only had this issue on the large rectangular complication using the Infograph Modular face.  All of the others seem to work well including the Modular face, but that seems to use a different large rectangular complication design (without the battery logo).

Love the App! Please keep up the good work!

Keith

**Comment by PhotonHebs** [comment]
I can confirm that this issue still persists in the beta version 1.1.7

Please take a look.  Let me know if I can help if you can’t repeat this bug.

**Comment by Kemmey** [comment]
Sorry for the radio silence. 

I've made a specific beta channel for this issue - if you'd like to join, do so here:

https://testflight.apple.com/join/2KESKzL7

The build removes all complications except the state complication. I'd be very interested to hear if the red icon still shows up then. 

Cheers,

Kim

**Comment by PhotonHebs** [comment]
Kim,

I have installed the build and reset. I will let you know how it goes.  Thank you 👍

Kind Regards

Keith

On 22 May 2021, at 20:39, Kim Hansen ***@***.***> wrote:

﻿

Sorry for the radio silence.

I've made a specific beta channel for this issue - if you'd like to join, do so here:

https://testflight.apple.com/join/2KESKzL7

The build removes all complications except the state complication. I'd be very interested to hear if the red icon still shows up then.

Cheers,

Kim

**Comment by chinedu40** [comment]
Thanks for this. I’ve just installed it. I needed to switch away from the state complication and then back first in order for it to work, but I’ll report back on how it goes

**Comment by SGS68** [comment]
Thanks. I’ve added it. Switched to different complication snd then added it back. It did as planned and only let me add state complication. Will test for a while. Appreciate you trying this.

V/r,
Shawn

Shawn Savchenko
Cell:  (919) 333.6285
***@***.***

**Comment by chinedu40** [comment]
Unfortunately after around 4 hours the issue reoccurred 

![image](https://user-images.githubusercontent.com/6845955/119253340-ba558a00-bba8-11eb-8b18-4486a828e682.jpg)

**Comment by PhotonHebs** [comment]
It worked well for a few hours (it seemed to last longer than normal but I could be wrong), then the issue occurred.

**Comment by chinedu40** [comment]
I installed the latest build. It unfortunately still doesn’t work and after a few hours just displays dashed lines when you raise your wrist and look at the complication.
![image](https://user-images.githubusercontent.com/6845955/119465932-0e44a800-bd3c-11eb-8d06-d408e712c16e.jpg)

**Comment by Kemmey** [comment]
How about last nights build - any change with that? Build 598?

**Comment by chinedu40** [comment]
Hi Kemmey, 

Unfortunately, same thing happens as with the previous builds

**Comment by Kemmey** [comment]
Thanks. 

But hey - now for the first time in months I'm able to reproduce!!!

I noticed your watch face colors were blue - I tried changing colors, and boom, I get the dashes!

I'll keep banging on this...

**Comment by chinedu40** [comment]
That's exciting! thanks for looking into this and i'm more than happy to continue testing

**Comment by Kemmey** [comment]
I was able to get it working again by removing some old watch faces. My theory was that maybe I had too many. Now it just works, and no matter how many faces I add, it still works :-(

So frustrating!

**Comment by Kemmey** [comment]
Aaaaand back to repro!

It must be something related to number of watch faces with watch for Tesla complications active. 

Do you guys have a lot of faces and have a lot of the complications active?

By a lot, I mean more than 10 watch for Tesla complications across 6+ faces?

**Comment by PhotonHebs** [comment]
I have the standard colour watch face on ‘infograph modular’.  It is my only watch face and the issue still occurs on build 598.  I will try different colours to see if that is a factor.

**Comment by Kemmey** [comment]
I think that might be a dead end - I'm shipping a new build as soon as it is done processing - it will be super ugly with just two lines of text for soc and range - but if my theory is correct, it might solve the problem. Then it'll be a matter of dialing it in for niceness...

**Comment by SGS68** [comment]
I only have 7 watch faces total. Tesla watch face is only in one of them the infograph modular snd using multi color.

Shawn

**Comment by Kemmey** [comment]
Build 599 is out. Fingers crossed this one works!!!

**Comment by PhotonHebs** [comment]
Have installed 599. Will let you know tomorrow how it turns out.  Thanks Kemmey.

**Comment by Kemmey** [comment]
Forgot to say - reboot of watch seems to be necessary to clear cache. Mines back to working with this build.

**Comment by PhotonHebs** [comment]
So far so good.  When I took it off charge this morning the dashes appeared (that hasn’t happened to me before - haven’t seen dashed lines at all).  

After I put the watch on my wrist and typed in my PIN the dashes disappeared. 

So far the 2 lines on the simple complication (battery% & miles) still shows on wrist raise - no sign of the red Tesla logo yet.

**Comment by chinedu40** [comment]
> So far so good. When I took it off charge this morning the dashes appeared (that hasn’t happened to me before - haven’t seen dashed lines at all).
> 
> After I put the watch on my wrist and typed in my PIN the dashes disappeared.
> 
> So far the 2 lines on the simple complication (battery% & miles) still shows on wrist raise - no sign of the red Tesla logo yet.

Can confirm that this was my experience also this morning. What is your current theory as to the cause of this problem?

**Comment by Kemmey** [comment]
Just to be clear - the dashes will show instead of the red logo, as the logo files have been completely erased from the build :-D 

I've been switching back and forth between old releases and this release - and can now somewhat consistently provoke the issue on old builds - but not this one. I'll add some more ui back and do a new build - fingers crossed the complication won't be stuck being ugly because of this bug...

**Comment by Kemmey** [comment]
My current theory is cpu limits on rendering time when using swiftui views in complications. 

So basically watchOS will determine that the complication was too expensive to render. This is where the bug lies - cause obviously it is not, as the inactive state correctly renders. Also this is not documented as a limitation anywhere, nor are any errors logged.

**Comment by Kemmey** [comment]
I have a good feeling for 601. I've been unable to repro with that build. 

Fingers crossed that applies universally

**Comment by PhotonHebs** [comment]
So far 599 & 601 have performed the same for me.  Dashes on initial reboot then it seems to be ok for wrist raise and lower.

**Comment by SGS68** [comment]
601 still working so far ok.

V/r,
Shawn

Shawn Savchenko
Cell:  (919) 333.6285
***@***.***

**Comment by gaardiolor** [comment]
I'm running 599, ok so far (~1 day, need at least one week to call it 'fixed'). I'm also having the dashes sometimes with other apps, when my watch is locked (and maybe until a few seconds after I unlock it). If you (in the latest builds) only see dashes in those conditions, I don't think it's related. Probably when the watch is locked, apps don't fetch complication updates, I think that's normal.

But, this shouldn't happen on an unlocked watch after the complication worked for some hours like with @chinedu40 in https://github.com/Kemmey/Kemmey-TeslaWatch-Public/issues/226#issuecomment-847669319 . That probably was an earlier build though.

**Comment by chinedu40** [comment]
So far version 604 is looking good. No dashed lines after waking up this morning. Will report back later this evening

**Comment by PhotonHebs** [comment]
601 has been perfect for a couple of days now. 
Installing 613 to see what that does :)

**Comment by Kemmey** [comment]
613 basically includes updated translations and a new image caching algorithm, that could potentially save some processing time, ie. battery :-D

**Comment by gaardiolor** [comment]
Still working :) it keeps autoupdating though (despite me disabling ‘automatic updates’ in testflight), I’m on 613 now. Not sure if the updates make this a 100% reliable test, but I’m hopeful :)

**Comment by PhotonHebs** [comment]
613 has worked great for me for over a day now.

thanks Kim, great work 👍

Now to see if I can trick my watch to summon/move my car with my phone inside; while I stand a bit further away than right next to it ;)

**Comment by Kemmey** [comment]
@PhotonHebs I removed summon again...

It's just too buggy to ship.

**Comment by gaardiolor** [comment]
613 still working :) Do you have enough info to fix this in the non-testflight version?

**Comment by Kemmey** [comment]
@gaardiolor Awesome!

Yeah, I think I'll consider this one solved. I'm preparing to ship 1.1.7 to the store - still missing the last few translations...

**Comment by gaardiolor** [comment]
Ai.. I have some bad news.

![EEE9AB97-62DB-4B84-85F8-8C106857CA65](https://user-images.githubusercontent.com/22920143/121815944-217ecf80-cc79-11eb-9878-73a758f8919e.png)
![3B6FC2FF-E48C-4E63-B11C-1680FE3A6A60](https://user-images.githubusercontent.com/22920143/121815945-22176600-cc79-11eb-9e66-c3ac1cfe76ef.png)

**Comment by Kemmey** [comment]
@gaardiolor Could you try 693?

**Comment by gaardiolor** [comment]
Assuming you mean 694? I don’t see 693 listed:
![6010E49F-3AB5-45A7-98A2-3B171B0E14BC](https://user-images.githubusercontent.com/22920143/121820172-a83fa680-cc91-11eb-9c52-8a34aaa04935.png)

**Comment by Kemmey** [comment]
@gaardiolor - yeah - 694 - sorry, I corrected it in the post just after sending :-D

**Comment by gaardiolor** [comment]
![IMG_3564](https://user-images.githubusercontent.com/22920143/122475084-6df14480-cfc4-11eb-8785-ca21aa7ef0eb.PNG)
![IMG_3565](https://user-images.githubusercontent.com/22920143/122475098-7184cb80-cfc4-11eb-9c56-91ba566e6ceb.PNG)

Sorry.. :(

**Comment by gaardiolor** [comment]
The issue also exists on 1.1.8 build 574

**Comment by Kemmey** [comment]
@gaardiolor sorry for the delayed response - been fighting another issue. 

Does this persist as prevalent as before or only once or twice? I've as usual been unable to reproduce, having that complication on my watch face for a few days now.

**Comment by gaardiolor** [comment]
Not sure, switched to another complication. Just switched back I’ll let you know!

**Comment by gaardiolor** [comment]
The issue just happened again (still on 1.1.8 build 754)

**Comment by gaardiolor** [comment]
Happened again today (same version). So it seems to happen less often, but still happens.. going to switch back to the small complication.
